### PR TITLE
Push request branch 1

### PR DESCRIPTION
**Pull request by rdiez**
This is my first pull request. Let's see if we can get such pull requests going.

What's the tab size you are using? I realised that the source is using tabs after my first commit attempt. I would actually prefer spaces.

I tried to fix some "Synchronization on non-final field" warnings too, but it turns out that you cannot have a variable marked as 'volatile' and 'final' at the same time.

I wonder whether those 'volatile' are really necessary. There is another NetBeans warning titled "volatile array field" on the same source code that suggests 'volatile' has no real impact.

### CODE REVIEW

**Comment by nyholku** [comment]
Hi,

thanks for PR.

The @Overrides are welcome but I don't want to introduce asserts into the code base, at least not at this point in time, not while we are working on something else, not in some random source file causing inconsistency. So let's leave that aside for now.

I will cherrypick the @Overrides over the weekend.

As to my tab size, that is 4.

Yes, there maybe a few superfluous volatiles but they are benign and it is better to be safe than sorry (having a field that should be volatile but is not can be a real pain to track down cause the problem may manifest itself only on some systems, with particular JVM, at rare and random time).

Lets concentrate on the real issues you (others) are having.

**Comment by mimmoz81** [comment]
Quick tip: there is a "throw" added to line 530 too, and it was probably missing in the master branch.
But I would do another PR for that fix..

File: src/jtermios/windows/JTermiosImpl.java [hunk 1]
```diff
@@ -499,15 +514,21 @@ public int write(int fd, byte[] buffer, int length) {
...
 							log = log && log(1, "write pending, cbInQue %d cbOutQue %d\n", port.m_COMSTAT.cbInQue, port.m_COMSTAT.cbOutQue);
 							continue;
 						}
 						if (!GetOverlappedResult(port.m_Comm, port.m_WrOVL, port.m_WrN, false))
 							port.fail();
-						if (port.m_WrN[0] != port.m_WritePending) // I exptect this is never going to happen, if it does
-							new RuntimeException("Windows OVERLAPPED WriteFile failed to write all, tried to write " + port.m_WritePending + " but got " + port.m_WrN[0]);
+						if (port.m_WrN[0] != port.m_WritePending) {
+							assert false;  // This case should never happen.
+							throw new RuntimeException("Windows OVERLAPPED WriteFile failed to write all, tried to write " + port.m_WritePending + " but got " + port.m_WrN[0]);
```
Comment (mimmoz81) [inline src/jtermios/windows/JTermiosImpl.java]: the "throw" is missing on the left side :O

**Comment by rdiez** [comment]
I think not including the asserts is a mistake. In my experience, they tend to help. They should not interfere with normal usage, for asserts are not normally enabled in the JVM.

Could you at least take the other changes in that commit? The new/reworked comments should help future coders, and the missing 'throw' is important.

File: src/jtermios/windows/JTermiosImpl.java (hunk 1 above)
Comment (nyholku) [reply src/jtermios/windows/JTermiosImpl.java]: Good catch!

**Comment by swarwick** [comment]
I believe that missing throw would have helped with my situation before we changed to FTDI chipset, so good catch.

**Pull request closed without merging**